examples/dynamic-projections-dask.py

```diff
@@ -0,0 +1,58 @@
+"""
+Using dask array operations, one can dynamically take arbitrary slices
+and computations of a source dask array and display the results in napari.
+When the computation takes one or more parameters, one can tie a UI to
+them using magicgui.
+"""
+
+import napari
+import dask.array as da
+from dask.array.lib.stride_tricks import sliding_window_view
+from skimage import data
+
+##############################################################################
+# Part 1: using code to view a specific value.
+
+blobs = data.binary_blobs(length=64, n_dim=3)
+blobs_dask = da.from_array(blobs, chunks=(1, 64, 64))
+
+# original shape [60, 1, 1, 5, 64, 64], slice out singleton axes
+blobs_dask_windows = sliding_window_view(
+    blobs_dask, window_shape=(5, 64, 64)
+)[:, 0, 0, ...]  
+blobs_sum = da.sum(blobs_dask_windows, axis=1)
+viewer = napari.view_image(blobs_sum)
+
+napari.run()
+
+##############################################################################
+# Part 2: using magicgui to vary the slice thickness.
+
+from magicgui import magicgui
+
+def sliding_window_mean(
+    arr: napari.types.ImageData, size: int = 1
+) -> napari.types.LayerDataTuple:
+    window_shape = (size,) + (arr.shape[1:])
+    arr_windows = sliding_window_view(arr, window_shape=window_shape)
+    arr_windows_1d = arr_windows[
+        (slice(None),) + (0,) * (arr.ndim - 1) + (Ellipsis,)
+    ]
```

Comment:
This line might benefit from a short comment, I'm a bit confused about the numpy magic :P